Controllers/OperationController.cs

```diff
@@ -99,6 +99,9 @@ public class OperationController : ApiController
         [Authorize]
         public async Task<HttpResponseMessage> Invoke(OperationInfo info)
         {
+            // escapeuristring does not escape '#' as it is a valid special character
+            // will not be legit in our case so escaping explicitly.
+            info.Url = Uri.EscapeUriString(info.Url).Replace("#", "%23");
```

Comment:
Maybe to play it safe we should just replace the `#`? I'm concerned that we could regress other scenarios by calling escape.